src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -60,18 +60,24 @@ class LedgerImpl(
     *         - [[BlockImportFailed]] - block failed to execute (when importing to top or reorganising the chain)
     */
   def importBlock(block: Block): BlockImportResult = {
-    //val validationResult: Either[Nothing, BlockHeader] = Right(block.header)
-    val validationResult = validators.blockHeaderValidator.validatePreImport(block.header, blockchain)
+    val validationResult = validateBlockBeforeExecution(block)
     validationResult match {
       case Left(error) =>
-        log.debug(s"Block(${block.header.number}: ${Hex.toHexString(block.header.hash.toArray)}) failed pre-import validation")
-        BlockImportFailed(error.toString)
+        val knownParent = blockchain.getBlockHeaderByHash(block.header.parentHash).isDefined || blockQueue.isQueued(block.header.parentHash)
```

Comment:
Minor: Won't we get a `HeaderParentNotFoundError` from `io.iohk.ethereum.ledger.LedgerImpl#validateBlockBeforeExecution` if parent is not present in the blockchain? That will prevent calling `getBlockHeaderByHash`

Reply:
Actually, it's checking it against queue too, right? `io.iohk.ethereum.ledger.LedgerImpl#getHeaderFromChainOrQueue`

Reply:
Yes, that check for parent was redundant, fixed